_posts/2020-05-11-microservices-on-axon.md

```diff
@@ -0,0 +1,198 @@
+---
+layout:    post
+title:     "Mikroserwisy na Axonie"
+date:      2020-05-11 08:00:00 +0100
+published: true
+author:    mkociszewski
+image:     
+tags:
+    - java
+    - microservices
+    - axon
+    - event-sourcing
+    - spring-boot
+---
+
+Znając definicję Event Sourcingu oraz korzyści, jakie nam zapewnia (dla przypomnienia polecam [**wpis Marcina poświęcony częściowo tej tematyce**]({% post_url 2018-11-15-czy-apache-kafka-nadaje-sie-do-event-sourcingu %})) warto rozważyć zastosowanie tego wzorca w swoim projekcie (oczywiście nie wszędzie się on nada).
+Osoby zainteresowane tematem z pewnością zostaną postawione przed wyborem technologii, w której rozpoczną implementację. 
+Niezależnie od języka programowania, można implementować CQRS oraz Event Sourcing samemu, od A-Z, jednakże byłoby to czasochłonne i może prowadzić do wielu błędów. 
+Alternatywą będzie zatem skorzystanie z gotowego frameworku, który od początku tworzony był z myślą o wspomnianych wzorcach (włączając w to mikroserwisy) - mowa tutaj o [**AxonFramework**](https://axoniq.io/).
+
+W tym wpisie przedstawię Axona i omówię wybory, przed którymi stałem w kontekście tego frameworka, oraz drogę migracji z monolitu do mikroserwisów wraz z problemami, na które się natknąłem.
+
+# Krótko o Axonie
+AxonFramework to... framework, który czerpie garściami z Domain Driven Design (które jest poza zakresem tego wpisu), wykorzystując również nomenklaturę panującą w tym podejściu, którą także będę się posługiwał w tym wpisie.
```

Comment:
usunąłbym te 3 kropki, wprowadzają zamieszanie

Reply:
to taka gra słów miała być z masła maślanego :sweat_smile: ale mogę to przeredagować, bo bez kropek to już zupełnie masło maślane